src/main/java/org/opentripplanner/updater/transportation_network_company/OAuthToken.java

```diff
@@ -0,0 +1,58 @@
+package org.opentripplanner.updater.transportation_network_company;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.time.Instant;
+import java.util.Date;
+
+/**
+ * Holds an OAuth access token and its expiration time for querying ride-hail APIs.
+ */
+public class OAuthToken {
+    public final String value;
+    private Date tokenExpirationTime;
```

Comment:
I'd recommend to use something from the `java.time` package, for example `Instant`. `java.util.Date` has so many pitfalls!